packages/@aws-cdk/aws-ecs/test/test.ecs-cluster.ts

```diff
@@ -274,8 +274,73 @@ export = {
           hardwareType: ecs.AmiHardwareType.GPU,
         }),
       });
+    }, /Amazon Linux does not support special hardware type/);
+
+    test.done();
+  },
+
+  "allows specifying windows image"(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'MyVpc', {});
+
+    const cluster = new ecs.Cluster(stack, 'EcsCluster', { vpc });
+    cluster.addCapacity('WindowsAutoScalingGroup', {
+      instanceType: new ec2.InstanceType('t2.micro'),
+      machineImage: new ecs.EcsOptimizedAmi({
+        windowsVersion: ecs.WindowsOptimizedVersion.SERVER_2019,
+      }),
     });
 
+    // THEN
+    expect(stack).to(haveResource("AWS::AutoScaling::LaunchConfiguration", {
```

Comment:
You should add an expectation on the parameter with the desired SSM path...

Reply:
Sorry, I don't understand your comment. I reused the existing HW AMI Type test: https://github.com/aws/aws-cdk/blob/47e81a63ff4b9d4326d56da2ac3b3fc04786cdd7/packages/%40aws-cdk/aws-ecs/test/test.ecs-cluster.ts#L252-L256

Reply:
This just shows the ImageId is a reference to a synthesized CloudFormation parameter that is added to your template. You should write your expectation against the default value of the parameter which will contain the SSM parameter name that you want.

What I usually do is `expect(stack).toMatch({})` and then I can see which parts of the stack to assert against.

Reply:
Good call, I was missing an assignation, thanks. 

I have an issue with the matching now. I tried changing the `MatchStyle`, but it doesn't seem to detect when I purposefully change the `Default` matched value.

Should I use an integration test instead, to match the whole stack?

Reply:
Yeah, I think there's a bug in the matcher (raise an issue?). What you could do in the meantime is as follows;

```ts
// you'll need an "App"
const app = new App();
const stack = new Stack(app, 'test');

// ...

// then
const assembly = app.synth();
const template = assembly.getStack(stack.stackName).template;
test.deepEqual(template.Parameters, { /* expectation here */ });
```

Reply:
Alright, thanks a lot. Should I also update the existing Linux AMI test? https://github.com/aws/aws-cdk/blob/5c01a39ceb3f61832c743795ee68205c414ce53c/packages/%40aws-cdk/aws-ecs/test/test.ecs-cluster.ts#L253-L257

If so, should I do it in another PR?

Reply:
Feel free to add to this PR

Reply:
Done. Thanks again for the help!